homework-submission/react-week4/src/app.js

```diff
@@ -0,0 +1,80 @@
+import React, { Component } from 'react';
+import Util from './utility';
+import AddingForm from './add-form';
+import Comment from './comment';
+import EditingForm from './editing-form';
+
+class App extends Component {
+  constructor(props) {
+    super(props);
+    this.state = { data: [], editing: "", error: "" }
+    this.handleCreate = this.handleCreate.bind(this);
+    this.handleEditing = this.handleEditing.bind(this);
+    this.handleUpdate = this.handleUpdate.bind(this);
+    this.handleDelete = this.handleDelete.bind(this);
+  }
+
+  refreshPage() {
+    Util.fetchList("https://hyf-react-api.herokuapp.com/blog/comments")
+      .then(data => this.setState({ data, editing: "" })).catch(err => {
+        this.handleError(err);
+      });
+  }
+
+  componentDidMount() { this.refreshPage() }
+
+  handleCreate(newItem) {
+    Util.createItem("https://hyf-react-api.herokuapp.com/blog/comments/create", newItem)
+      .then(res => this.refreshPage()).catch(err => {
```

Comment:
you are fetching the all data, just for one added item and that's not good
try to find another solution